lib/view_component/cacheable.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module ViewComponent::Cacheable
+  extend ActiveSupport::Concern
+
+  included do
+    class_attribute :__vc_cache_dependencies, default: [:format, :__vc_format, :identifier]
```

Comment:
Should this be a Set so that we have automatic deduplication? This would allow that one calls `cache_on :current_locale` in the `ApplicationComponent`. Subsequent calls to `cache_on` would not lead to duplicates in `__vc_cache_dependencies`.

Side-note: `cache_on` with lambda support would also be interesting. E.g. `cache_on -> lambda { I18n.current_locale }`